go/processor/geth/processor_test.go

```diff
@@ -94,3 +97,123 @@ func TestGethProcessor_ConfigAddsStateContract(t *testing.T) {
+	}
+	if msg.GasLimit != uint64(tx.GasLimit) {
+		t.Errorf("GasLimit mismatch: got %d, want %d", msg.GasLimit, tx.GasLimit)
+	}
+	if msg.GasPrice.Cmp(gasPrice.ToBig()) != 0 {
+		t.Errorf("GasPrice mismatch: got %v, want %v", msg.GasPrice, gasPrice.ToBig())
+	}
+	if msg.GasFeeCap.Cmp(tx.GasFeeCap.ToBig()) != 0 {
+		t.Errorf("GasFeeCap mismatch: got %v, want %v", msg.GasFeeCap, tx.GasFeeCap.ToBig())
+	}
+	if msg.GasTipCap.Cmp(tx.GasTipCap.ToBig()) != 0 {
+		t.Errorf("GasTipCap mismatch: got %v, want %v", msg.GasTipCap, tx.GasTipCap.ToBig())
+	}
+	if !bytes.Equal(msg.Data, tx.Input) {
+		t.Errorf("Input mismatch: got %x, want %x", msg.Data, tx.Input)
+	}
+	if msg.BlobGasFeeCap.Cmp(tx.BlobGasFeeCap.ToBig()) != 0 {
+		t.Errorf("BlobGasFeeCap mismatch: got %v, want %v", msg.BlobGasFeeCap, tx.BlobGasFeeCap.ToBig())
+	}
+	if len(msg.BlobHashes) != 1 || msg.BlobHashes[0] != blobHashes[0] {
+		t.Errorf("BlobHashes mismatch: got %v, want %v", msg.BlobHashes, blobHashes)
+	}
+}
+
+func TestTransactionToMessage_AccessList(t *testing.T) {
+	tx := tosca.Transaction{
+		Sender: tosca.Address{0x01},
+		AccessList: []tosca.AccessTuple{
+			{
+				Address: tosca.Address{0x03},
+				Keys:    []tosca.Key{{0x42}, {0x43}},
+			},
+		},
+	}
+	gasPrice := tosca.NewValue(1)
+	msg := transactionToMessage(tx, gasPrice, nil)
+	if len(msg.AccessList) != 1 {
+		t.Fatalf("expected 1 access tuple, got %d", len(msg.AccessList))
+	}
+	if msg.AccessList[0].Address != common.Address(tx.AccessList[0].Address) {
+		t.Errorf("AccessList address mismatch")
+	}
+	if len(msg.AccessList[0].StorageKeys) != 2 {
+		t.Errorf("expected 2 storage keys, got %d", len(msg.AccessList[0].StorageKeys))
+	}
+	if msg.AccessList[0].StorageKeys[0] != common.Hash(tx.AccessList[0].Keys[0]) {
+		t.Errorf("StorageKey[0] mismatch")
+	}
+	if msg.AccessList[0].StorageKeys[1] != common.Hash(tx.AccessList[0].Keys[1]) {
+		t.Errorf("StorageKey[1] mismatch")
+	}
+}
+
+func TestTransactionToMessage_AuthorizationList(t *testing.T) {
+	tx := tosca.Transaction{
+		Sender: [20]byte{0x01},
+		AuthorizationList: []tosca.SetCodeAuthorization{
+			{
+				ChainID: tosca.Word(tosca.NewValue(0x01)),
+				Address: tosca.Address{0x02},
+				Nonce:   7,
+				V:       42,
+				R:       tosca.Word(tosca.NewValue(0x03)),
+				S:       tosca.Word(tosca.NewValue(0x04)),
+			},
+		},
+	}
+	gasPrice := tosca.NewValue(1)
+	msg := transactionToMessage(tx, gasPrice, nil)
+	if len(msg.SetCodeAuthorizations) != 1 {
+		t.Fatalf("expected 1 authorization, got %d", len(msg.SetCodeAuthorizations))
+	}
+	auth := msg.SetCodeAuthorizations[0]
+	if auth.ChainID.Cmp(uint256.NewInt(1)) != 0 {
+		t.Errorf("ChainID mismatch: got %v", auth.ChainID)
+	}
+	if auth.Address != common.Address(tx.AuthorizationList[0].Address) {
+		t.Errorf("Address mismatch")
+	}
+	if auth.Nonce != tx.AuthorizationList[0].Nonce {
+		t.Errorf("Nonce mismatch")
+	}
+	if auth.V != tx.AuthorizationList[0].V { //nolint:all
```

Comment:
nit: For R and S you are avoiding the deprecated linter warning by comparing against the actual value, not the value from the input. So maybe here (unless you remove the deprecated state as mentioned in another comment):

```suggestion
	if auth.V != 42 {
```